cmake/bitcoin-config.h.in

```diff
@@ -63,6 +180,9 @@
 /* Define to the version of this package. */
 #define PACKAGE_VERSION "@PROJECT_VERSION@"
 
+/* Define to 1 if strerror_r returns char *. */
+#cmakedefine STRERROR_R_CHAR_P 1
+
```

Comment:
Going through all the symbols defined in here and could not find the following builtins from our current bitcoin-config.h.in:
`HAVE_BUILTIN_CLZL` `HAVE_BUILTIN_CLZLL` `HAVE_BUILTIN_MUL_OVERFLOW`.

The crypto intrinsics are accounted for in the next PR, but I couldn't find `HAVE_TIMINGSAFE_BCMP`.

I also could not find `HAVE_EVHTTP_CONNECTION_GET_PEER_CONST_CHAR` and `HAVE_THREAD_LOCAL`.

My guess is I'm missing something here. The other symbols not carried over from the old config are indeed unused.

Reply:
@TheCharlatan 

Thank you so much!

> Going through all the symbols defined in here and could not find the following builtins from our current bitcoin-config.h.in: `HAVE_BUILTIN_CLZL` `HAVE_BUILTIN_CLZLL` `HAVE_BUILTIN_MUL_OVERFLOW`.

Fixed.

> The crypto intrinsics are accounted for in the next PR, but I couldn't find `HAVE_TIMINGSAFE_BCMP`.

Going to add it to next PR.

> I also could not find `HAVE_EVHTTP_CONNECTION_GET_PEER_CONST_CHAR` and `HAVE_THREAD_LOCAL`.

Same. 

`HAVE_EVHTTP_CONNECTION_GET_PEER_CONST_CHAR` should be an amend of `pkg_check_modules(libevent)`, and `HAVE_THREAD_LOCAL` should be added to `find_package(Threads)`.

Reply:
For `HAVE_THREAD_LOCAL` see `target_compile_features( ... cxx_thread_local ... )`.

Reply:
The [diff](https://github.com/hebasto/bitcoin/compare/e6ebaa0a39fcbeea506d2cb20ad94b8b1476c61e..9e218cfca5b863ced0764da7af8b202965c0cd34) you pushed doesn't introduce `HAVE_BUILTIN_MUL_OVERFLOW`. Is this left for later?

Reply:
> The [diff](https://github.com/hebasto/bitcoin/compare/e6ebaa0a39fcbeea506d2cb20ad94b8b1476c61e..9e218cfca5b863ced0764da7af8b202965c0cd34) you pushed doesn't introduce `HAVE_BUILTIN_MUL_OVERFLOW`. Is this left for later?

Yes, it is because it is a fuzz binary specific.

Reply:
Perfect, from my point of view all symbols are now accounted for :)

Reply:
> For `HAVE_THREAD_LOCAL` see `target_compile_features( ... cxx_thread_local ... )`.

[Done](https://github.com/hebasto/bitcoin/pull/7#issuecomment-1439089362) in #7.